src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/AddUsingsHelper.cs

```diff
@@ -137,21 +141,17 @@ public static WorkspaceEdit CreateAddUsingWorkspaceEdit(string @namespace, TextD
         };
     }
 
-    private static async Task<IEnumerable<string>> FindUsingDirectiveStringsAsync(SourceText originalCSharpText, CancellationToken cancellationToken)
+    public static async Task<ImmutableArray<string>> FindUsingDirectiveStringsAsync(SyntaxTree syntaxTree, CancellationToken cancellationToken)
     {
-        var syntaxTree = CSharpSyntaxTree.ParseText(originalCSharpText, cancellationToken: cancellationToken);
         var syntaxRoot = await syntaxTree.GetRootAsync(cancellationToken).ConfigureAwait(false);
 
         // We descend any compilation unit (ie, the file) or and namespaces because the compiler puts all usings inside
         // the namespace node.
         var usings = syntaxRoot.DescendantNodes(n => n is BaseNamespaceDeclarationSyntax or CompilationUnitSyntax)
-            // Filter to using directives
             .OfType<UsingDirectiveSyntax>()
-            // Select everything after the initial "using " part of the statement, and excluding the ending semi-colon. The
-            // semi-colon is valid in Razor, but users find it surprising. This is slightly lazy, for sure, but has
-            // the advantage of us not caring about changes to C# syntax, we just grab whatever Roslyn wanted to put in, so
-            // we should still work in C# v26
-            .Select(u => u.ToString()["using ".Length..^1]);
+            .Select(u => u.Name)
+            .WhereNotNull()
+            .SelectAsArray(n => n.ToString());
```

Comment:
Okay, I think this is your follow up, https://github.com/dotnet/razor/issues/11168. I'll leave the comment though, because that issue mentions the Razor tree being inconsistent, but I'm pretty sure which ever solution we come up with for that would also have to be consistent with this.

Perhaps the answer is just to use the string representation of the entire directive, sans `@` in Razor and `;` in C#